utils/validation.js

```diff
@@ -178,6 +178,70 @@ class Validation {
 
         return true;
     }
+
+    /**
+     * THIS IS SLOW. REWRITE THIS USING A SET INSTEAD
+     * Check whether IDs are unique in the given array.
```

Comment:
You don't have to do it in this PR but you definitely should do this. If you haven't already you should open an issue about it (and possibly refer to that issue here)

Reply:
I don't believe I'm using that function anywhere. In fact there are a few functions in validation that I was going to use for Architecture, but won't need to anymore. I'll open an issue to clean up the validation class.

Reply:
#41 deals with this.